File: aoc-2016/src/test/java/net/bbelovic/adventofcode/day1/Day1PuzzlePart1Test.java
package net.bbelovic.adventofcode.day1;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day1PuzzlePart1Test {
    @ParameterizedTest
    @MethodSource("testData")
    void should_compute_distance_from_start(String input, int expectedDistance) {
        long actualDistance = new Day1PuzzlePart1().solve(input);
        assertEquals(expectedDistance, actualDistance);
    }

    private static Stream<Arguments> testData() {
        return Stream.of(Arguments.of("R2, L3", 5),
                Arguments.of("R2, R2, R2", 2), Arguments.of("R5, L5, R5, R3", 12));
    }
}

File: aoc-2018/src/test/kotlin/net/bbelovic/adventofcode/year2018/day1/Day1PuzzlePart1Test.kt
package net.bbelovic.adventofcode.year2018.day1

import net.bbelovic.adventofcode.Puzzle
import net.bbelovic.adventofcode.utils.dsl.testDataDsl
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.MethodSource

class Day1PuzzlePart1Test {
    @ParameterizedTest
    @MethodSource("testData")
    fun should_compute_final_frequency_from_frequency_changes(input: String, expectedFrequency: Long) {
        val puzzle: Puzzle<String, Long> = Day1PuzzlePart1()
        val actualFrequency = puzzle.solve(input)
        assertEquals(expectedFrequency, actualFrequency)
    }

    companion object {
        @JvmStatic
        fun testData() = testDataDsl {
            testArguments ("+1, -2, +3, +1", 3L)
            testArguments ("+1, +1, +1", 3L)
            testArguments ("+1, +1, -2", 0L)
            testArguments ("-1, -2, -3", -6L)
        }
    }
}